src/coreclr/src/utilcode/pedecoder.cpp

```diff
@@ -1770,20 +1770,31 @@ void PEDecoder::LayoutILOnly(void *base, BOOL allowFullPE) const
                            PAGE_READONLY, &oldProtection))
         ThrowLastError();
 
-    // Finally, apply proper protection to copied sections
-    section = sectionStart;
-    while (section < sectionEnd)
+    // Finally, apply proper protection to copied sections    
+    for (section = sectionStart; section < sectionEnd; section++)
     {
         // Add appropriate page protection.
-        if ((section->Characteristics & VAL32(IMAGE_SCN_MEM_WRITE)) == 0)
+#if defined(CROSSGEN_COMPILE) || defined(TARGET_UNIX)
```

Comment:
is `CROSSGEN_COMPILE` R2R?

Reply:
I think CROSSGEN_COMPILE is when coreclr code is used as a part of crossgen tool. 
In that case the goal is introspection and jitting ahead of time, not running, so there is a good deal of ifdefs in cases which are specific to actually running code.